client/allocrunner/consul_hook.go

```diff
@@ -98,19 +98,25 @@ func (h *consulHook) Prerun() error {
 }
 
 func (h *consulHook) prepareConsulTokensForTask(job *structs.Job, task *structs.Task, tgName string, tokens map[string]map[string]string) error {
+	var consulClusterName string
+	if task.Consul != nil && task.Consul.Cluster != "" {
+		consulClusterName = task.Consul.Cluster
+	} else {
+		consulClusterName = structs.ConsulDefaultCluster
+	}
```

Comment:
I think this should be handled way earlier than in the alloc runner. For Vault it's set on job registration:
https://github.com/hashicorp/nomad/blob/3633ca0f8ca4c0a7d4ecb43840e1240c10f45116/nomad/job_endpoint_hook_vault_ce.go#L44-L55

But it could be handled in a separate PR as it requires some ENT changes.